lib/pure/sugar.nim

```diff
@@ -54,6 +54,8 @@ proc createProcType(p, b: NimNode): NimNode {.compileTime.} =
 
 macro `=>`*(p, b: untyped): untyped =
   ## Syntax sugar for anonymous procedures. It also supports pragmas.
+  ##
+  ## In addition, this can be used to define named procs.
```

Comment:
```suggestion
  ## Syntax sugar for anonymous and named procedures.
  ## It also supports multi-line procedures, and multiple expressions or statements.
  ## It can use pragmas and generics, has no extra overhead over a normal procedure.
```

:thinking:

Reply:
> has no extra overhead over a normal procedure

how so?

Reply:
Meaning that expands to a normal procedure during compilation.

Some langs have limited arrows or anon or lambdas, eggsample:
- Python can do multi-line lambda, but using `;` most linters and formatters complain about `;`.
- Even if multi-line, must be single-expression so you have to write it kinda weird.
- Must be expression, complains with statements.
- Known to be slower than normal functions.

I did not say "generates the same code", I say theres no relevant extra overhead over a `proc`.
Ignore the suggestion if you think is too verbose.